packages/block/types.ts

```diff
@@ -0,0 +1,81 @@
+export type { VNode, VElement } from '../million/types';
+export type Props = Record<string, any>;
+
+export class Hole {
+  key: string;
+  constructor(key: string) {
+    this.key = key;
+  }
+}
+
+export abstract class AbstractBlock {
+  root?: HTMLElement;
+  edits?: Edit[];
+  el?: HTMLElement;
+  _parent?: HTMLElement | null;
+  props?: Props | null;
+  key?: string;
+  cache?: Map<number, HTMLElement>;
+  abstract patch(block: AbstractBlock): HTMLElement;
+  abstract mount(parent?: HTMLElement, refNode?: Node | null): HTMLElement;
+  abstract move(block: AbstractBlock | null, refNode: Node | null): void;
+  abstract remove(): void;
+  abstract toString(): string;
+  shouldUpdate?(oldProps: Props, newProps: Props): boolean;
+  get parent(): HTMLElement | null | undefined {
+    return this._parent;
+  }
+}
+
+export const enum EditType {
+  Attribute,
+  Block,
+  Child,
+  Event,
+}
```

Comment:
`unbuild` uses `esbuild` under the hood, which doesn't support cross-file inlining. So here is what `unbuild` outputs:

```js
var EditType = /* @__PURE__ */ ((EditType2) => {
  EditType2[EditType2["Attribute"] = 0] = "Attribute";
  EditType2[EditType2["Block"] = 1] = "Block";
  EditType2[EditType2["Child"] = 2] = "Child";
  EditType2[EditType2["Event"] = 3] = "Event";
  return EditType2;
})(EditType || {});
```

You can verify it by running `pnpm run build` and checking the `dist` folder.

This could drastically reduce the performance.

Before `esbuild` supports cross-file inlining, we should avoid using `const enum`. I propose this instead:

```js
export const EditTypeAttribute = 1
export const EditTypeBlock = 2
export const EditTypeChild = 3
export const EditTypeEvent = 4
```

Reply:
Got it, changing now.